geograypher/cameras/cameras.py

```diff
@@ -930,6 +934,229 @@ def get_camera_locations(self, **kwargs):
+                version of an image, we need to track that mapping seperately
+                because downsampling affects the pixel to pixel mapping.
+                Include the image scale (0-1 float) along with the distortion
+                parameters to make a key.
+
+        Returns:
+            A repeatable string key based on the distortion values.
+        """
+        keys = sorted(parameters.keys())
+        strings = [f"{key}:{parameters[key]:.8f}" for key in keys] + [
+            f"image_scale:{image_scale:.8f}"
+        ]
+        return "|".join(strings)
+
+    def make_distortion_map(
+        self,
+        camera: PhotogrammetryCamera,
+        inversion_downsample: int,
+        image_scale: float = 1.0,
+    ) -> None:
+        """
+        Cache a map connecting locations in one image to locations in another.
+        The basic construction is the pixel position of the map is the position
+        in the destination image, and the value of the map is the position in
+        the source image. Therefore if location [20, 30] has value [22.2, 28.4],
+        it means that the destination image pixel [20, 30] will be sampled from
+        the source image at pixel [22.2, 28.4] (the sampler can choose to snap
+        to the closest integer value or interpolate nearby pixels).
+
+        Arguments:
+            camera (PhotogrammetryCamera): Camera with parameters that
+                define the warp process. These include image size, principal
+                point, focal length, and distortion parameters.
+            inversion_downsample (int): Downsample the inverse map process in
+                order to reduce computation, at high res the resulting map
+                trafeoffs should be minimal.
+            image_scale: (float, optional) 0-1 fraction of the original image
+                size, used when warping/dewarping downsampled images. See
+                pix2face render_img_scale for an example.
+
+        Caches:
+            In self._maps_ideal_to_warped, stores a map of the structure
+                discussed above, keyed by self.distortion_key(params) so it can
+                be accessed for cameras using the same params.
+            In self._maps_warped_to_ideal, stores an inverted version
+        """
+
+        # Sample over the ideal pixels, shape (H, W)
+        im_h, im_w = camera.image_size
+        if np.isclose(image_scale, 1.0):
+            h_range = np.arange(im_h)
+            w_range = np.arange(im_w)
+        else:
+            # In order to get a distortion map for a downsampled image, you
+            # need to run the ideal_to_warped equation over the same original
+            # range (0...im_h) because the radius of that range relates
+            # directly to the warping. However, do it in fewer steps (step
+            # size > 1) to reflect the downsampling.
+            h_range = np.arange(im_h, step=1 / image_scale)[: int(im_h * image_scale)]
```

Comment:
Is the indexing at the end to deal with floating point math potentially having an extra sample? 

One unrelated comment: this could potentially be slightly more accurate if you shifted these samples by `1 / (2 * image_scale)` so the sample points fall at the center of the steps rather than the top left bound.